src/components/ButtonPanel.js

```diff
@@ -1,39 +1,58 @@
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+/* eslint-disable react/jsx-filename-extension */
 /* eslint-disable react/prefer-stateless-function */
 import React from 'react';
 import Button from './Button';
 
 class ButtonPanel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
```

Comment:
With respect to your problem upon submitting this PULL  REQUEST, wondering where you will get the `calculatorData`, this is the data here initiated in the `this.state` which you'll use in collaboration with `buttonName` as parameter of `handleClick()` function

Reply:
this means inside the handleClick, you will have to call upon `calculate()`, and use it's results(which is also an object) to UPDATE the `this.state`